### PR TITLE
Honor RecursiveDir when used on packed items in framework packs

### PR DESCRIPTION
**Pull request by ericstj**
When packaging Analyzers in the ref-pack we were setting PackagePath to the full path inside the package (by appending `%(RecursiveDir)`) then Nuget was appending `%(RecursiveDir)` again during pack.  Avoid this by removing the expansion in our own items and fixing the `CreateFrameworkList` task to append `%(RecurisveDir)` in a similar manner to NuGet.

Addresses #34272

### CODE REVIEW

**Comment by captainsafia** [comment]
Thanks for digging into this. Let's try undoing the workarounds made in the dependency update PR to see if this works. I can undo the changes made in that PR if you add me as a collaborate to your fork, @ericstj.

**Comment by ericstj** [comment]
I've added you, @captainsafia

**Comment by captainsafia** [comment]
Looks like there is still a failure related to the logging generator:

```
CSC(0,0): error CS0006: (NETCORE_ENGINEERING_TELEMETRY=Build) Metadata file 'D:\workspace\_work\1\s\.dotnet\packs\Microsoft.AspNetCore.App.Ref\6.0.0-ci\analyzers/dotnet/cs/Microsoft.Extensions.Logging.Generators.dll' could not be found
```

The error occurs when building the `InteropWebsite` project on Helix.

**Comment by ericstj** [comment]
It looks like it's running from packs copied to the repo local `.dotnet/packs` folder.   I suspect there is part of AspNet infra that's copying things to that location.  It must be copying the framework list but not the rest of the ref pack.

**Comment by ericstj** [comment]
I see, these are additional targets in the ref-pack project.

File: eng/tools/RepoTasks/CreateFrameworkListFile.cs [hunk 1]
```diff
@@ -125,5 +125,17 @@ public override bool Execute()
 
             return !Log.HasLoggedErrors;
         }
+        private static string GetPackagePath(ITaskItem item)
```
Comment (dougbu) [inline eng/tools/RepoTasks/CreateFrameworkListFile.cs:128]: nit:
```suggestion

        private static string GetPackagePath(ITaskItem item)
```

**Review by dougbu — APPROVED** [review]

**Comment by captainsafia** [comment]
Merging to unblock SDK insertion.

**Pull request merged**